### PR TITLE
Stamp all tools with version information.

### PR DESCRIPTION
**Pull request by Alan-Jowett**
Resolves: #3791 

## Description

This pull request enhances the build configuration for several tools by adding resource compilation steps and defining resource files. The changes ensure that resource files are included and properly referenced in the build process.

The resource files contain both the release version number as well as the git commit ID used to build this file.

### Build Configuration Enhancements:

* [`tools/bpf2c/bpf2c.vcxproj`](diffhunk://#diff-6f0c1c66f8f11519ef4fe6bd91ca59149585e559f19e0a1e0ab42f89876ceceeR120-R122): Added `ResourceCompile` steps with `AdditionalIncludeDirectories` for various build configurations and included `ebpf_resource.rc` in the project. [[1]](diffhunk://#diff-6f0c1c66f8f11519ef4fe6bd91ca59149585e559f19e0a1e0ab42f89876ceceeR120-R122) [[2]](diffhunk://#diff-6f0c1c66f8f11519ef4fe6bd91ca59149585e559f19e0a1e0ab42f89876ceceeR138-R140) [[3]](diffhunk://#diff-6f0c1c66f8f11519ef4fe6bd91ca59149585e559f19e0a1e0ab42f89876ceceeR156-R158) [[4]](diffhunk://#diff-6f0c1c66f8f11519ef4fe6bd91ca59149585e559f19e0a1e0ab42f89876ceceeR177-R179) [[5]](diffhunk://#diff-6f0c1c66f8f11519ef4fe6bd91ca59149585e559f19e0a1e0ab42f89876ceceeR198-R200) [[6]](diffhunk://#diff-6f0c1c66f8f11519ef4fe6bd91ca59149585e559f19e0a1e0ab42f89876ceceeR298-R300)
* [`tools/bpf2c/bpf2c.vcxproj.filters`](diffhunk://#diff-39fef2814c965c6841d9e06507d7ef16730669311736e8f09adc77a542986f13R67-R71): Added a filter for resource files.
* [`tools/bpftool/bpftool.vcxproj`](diffhunk://#diff-12224e44cd315f897ad8409b2c53884280ce1ab57cbb30ceb3cb0e510ed1f6dfR157-R164): Added `ResourceCompile` steps with `AdditionalIncludeDirectories` for various build configurations and included `ebpf_resource.rc` in the project.
* [`tools/bpftool/bpftool.vcxproj.filters`](diffhunk://#diff-167039d353f498bbd668956d0ba10f033365e532368c79a2a3b0bf1769d5863cR133-R137): Added a filter for resource files.
* [`tools/export_program_info/export_program_info.vcxproj`](diffhunk://#diff-89f2fb43211d88943131c99d17f4ea3f99d32aed85791980c0a6ec2fba5ac73eR258-R266): Added `ResourceCompile` steps with `AdditionalIncludeDirectories` for various build configurations and included `ebpf_resource.rc` in the project.
* [`tools/export_program_info/export_program_info.vcxproj.filters`](diffhunk://#diff-bb0a605a33f822336eafd1a6da9e9e9d03acd896994a96bdcb02260e2662c475R40-R44): Added a filter for resource files.
* [`undocked/tools/export_program_info_sample/export_program_info_sample.vcxproj`](diffhunk://#diff-f8657953e2f7ab6d37480f69d359beee7159212559559d94a0bf9821ecf07e15R136-R138): Added `ResourceCompile` steps with `AdditionalIncludeDirectories` for various build configurations and included `ebpf_resource.rc` in the project. [[1]](diffhunk://#diff-f8657953e2f7ab6d37480f69d359beee7159212559559d94a0bf9821ecf07e15R136-R138) [[2]](diffhunk://#diff-f8657953e2f7ab6d37480f69d359beee7159212559559d94a0bf9821ecf07e15R162-R164) [[3]](diffhunk://#diff-f8657953e2f7ab6d37480f69d359beee7159212559559d94a0bf9821ecf07e15R188-R190) [[4]](diffhunk://#diff-f8657953e2f7ab6d37480f69d359beee7159212559559d94a0bf9821ecf07e15R216-R218) [[5]](diffhunk://#diff-f8657953e2f7ab6d37480f69d359beee7159212559559d94a0bf9821ecf07e15R244-R246) [[6]](diffhunk://#diff-f8657953e2f7ab6d37480f69d359beee7159212559559d94a0bf9821ecf07e15R260-R262)
* [`undocked/tools/export_program_info_sample/export_program_info_sample.vcxproj.filters`](diffhunk://#diff-7b5f95b9def7d33530830aba8d35e710b2f699cae87027bc4443000efaae31caR34-R38): Added a filter for resource files.

### Resource Definitions:

* [`tools/bpf2c/resource.h`](diffhunk://#diff-a94e7ce6695f44d0998db5b165901bc43ce354470bbdf4b29311fcd4aaab1821R1-R5): Added resource definitions for `bpf2c.exe`.
* [`tools/bpftool/resource.h`](diffhunk://#diff-2271b61769776d565a9cba5844aed4d7ba7799dcfff6246a120213ef9c1c9f87R1-R5): Added resource definitions for `bpftool.exe`.
* [`tools/export_program_info/resource.h`](diffhunk://#diff-1d748f6c52e4f48e09c82d3ed1f400ff1b8932c814fc59dc580dbe455cda2858R1-R5): Added resource definitions for `export_program_info.exe`.
* [`undocked/tools/export_program_info_sample/resource.h`](diffhunk://#diff-35e47a4d359ffcdaf6aaf4964291a27829e1f4fcf8d5647dc30da5c6669e4f17R1-R5): Added resource definitions for `export_program_info_sample.exe`.

## Testing

CI/CD

## Documentation

No.

## Installation

No.

### CODE REVIEW

File: tools/bpftool/bpftool.vcxproj [hunk 1]
```diff
@@ -154,6 +154,14 @@
       <Project>{75fe223a-3e45-4b0e-a2e8-04285e52e440}</Project>
     </ProjectReference>
   </ItemGroup>
+  <ItemGroup>
+    <ResourceCompile Include="..\..\resource\ebpf_resource.rc">
+      <AdditionalIncludeDirectories Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">$(SolutionDir)include</AdditionalIncludeDirectories>
```
Comment (Austin-Lamb) [inline tools/bpftool/bpftool.vcxproj:159]: It'd be nice if the `<AdditionalIncludeDirectories>` was just specified once, instead of once per config/platform since it's the same for all of them anyway, just to have less MSBuild to read and maintain

File: tools/bpf2c/bpf2c.vcxproj [hunk 2]
```diff
@@ -132,6 +135,9 @@
     <PreBuildEvent>
       <Command>powershell -NonInteractive -ExecutionPolicy Unrestricted $(SolutionDir)scripts\Process-File.ps1 -InputFile Convert-BpfToNative.ps1.template -OutputFile $(OutDir)Convert-BpfToNative.ps1 -ConfigFile replacements.json</Command>
     </PreBuildEvent>
+    <ResourceCompile>
```
Comment (Austin-Lamb) [inline tools/bpf2c/bpf2c.vcxproj:138]: Consider putting the `<ResourceCompile>` additional include directories into `Directory.Build.props` so it happens for everything in the repo including things that get added in the future, rather than putting in each project individually?

File: tools/export_program_info/export_program_info.vcxproj [hunk 3]
```diff
@@ -255,5 +255,14 @@ $(OutputPath)export_program_info.exe &gt; $(OutputPath)export_program_info.log
       <Project>{aa933b9f-b5d8-4aa8-ac18-98fe1a161e8a}</Project>
     </ProjectReference>
   </ItemGroup>
+  <ItemGroup>
+    <ResourceCompile Include="..\..\resource\ebpf_resource.rc">
+      <AdditionalIncludeDirectories Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">$(SolutionDir)include</AdditionalIncludeDirectories>
```
Comment (Austin-Lamb) [inline tools/export_program_info/export_program_info.vcxproj:260]: Same here that it'd be nice to repeat this less often - either in `Directory.Build.props` or just once without a condition for each config/platform

File: ebpfcore/usersim/EbpfCore_Usersim.vcxproj [hunk 4]
```diff
@@ -132,10 +132,6 @@
     <DebuggerFlavor>DbgengKernelDebugger</DebuggerFlavor>
   </PropertyGroup>
   <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Release|x64'">
-    <ResourceCompile>
-      <AdditionalIncludeDirectories>%(AdditionalIncludeDirectories);$(DDK_INC_PATH);$(SolutionDir)include</AdditionalIncludeDirectories>
-      <PreprocessorDefinitions>%(PreprocessorDefinitions);BINARY_COMPATIBLE=0;NT;UNICODE;_UNICODE;NDIS60;POOL_NX_OPTIN_AUTO</PreprocessorDefinitions>
```
Comment (Austin-Lamb) [inline ebpfcore/usersim/EbpfCore_Usersim.vcxproj:137]: This one used to pass some different preprocessor definitions and the DDK include path - is it ok for it to inherit the common stuff and not get those?

Comment (Austin-Lamb) [reply ebpfcore/usersim/EbpfCore_Usersim.vcxproj:137]: I guess a few of them passed these in - but it doesn't seem like they were used, so if you agree that they weren't used then it's great that they standardize on the same set of defines for resources for the whole repo.

**Review by Austin-Lamb — APPROVED** [review]
:shipit:

File: ebpfcore/usersim/EbpfCore_Usersim.vcxproj (hunk 4 above)
Comment (Alan-Jowett) [reply ebpfcore/usersim/EbpfCore_Usersim.vcxproj:137]: Yes, they appear to be unused.

File: scripts/setup_build/setup_build.vcxproj.filters [hunk 5]
```diff
@@ -78,4 +78,4 @@
       <Filter>Source Files</Filter>
     </CustomBuild>
   </ItemGroup>
-</Project>
+</Project>
```
Comment (dthaler) [inline scripts/setup_build/setup_build.vcxproj.filters:81]: Why this change?
Same question on other .filters files in this PR.

File: tests/socket/resource.h [hunk 6]
```diff
@@ -0,0 +1,5 @@
+// Copyright (c) eBPF for Windows contributors
+// SPDX-License-Identifier: MIT
+
+#define EBPF_FILE_DESCRIPTION "Socket test"
```
Comment (dthaler) [inline tests/socket/resource.h:4]:
```suggestion
#define EBPF_FILE_DESCRIPTION "Socket tests"
```
For consistency with the file name, and also with file descriptions on the other test executables

File: tools/bpftool/bpftool.vcxproj [hunk 7]
```diff
@@ -279,4 +279,4 @@
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
   <ImportGroup Label="ExtensionTargets">
   </ImportGroup>
-</Project>
+</Project>
```
Comment (dthaler) [inline tools/bpftool/bpftool.vcxproj:282]: Why this change?

File: tools/onebranch/onebranch.vcxproj.filters [hunk 8]
```diff
@@ -17,4 +17,9 @@
       <Extensions>rc;ico;cur;bmp;dlg;rc2;rct;bin;rgs;gif;jpg;jpeg;jpe;resx;tiff;tif;png;wav;mfcribbon-ms</Extensions>
     </Filter>
   </ItemGroup>
+  <ItemGroup>
+    <ResourceCompile Include="$(SolutionDir)resource\ebpf_resource.rc">
```
Comment (dthaler) [inline tools/onebranch/onebranch.vcxproj.filters:21]: Why does this file need ebpf_resource.rc added whereas the others needed it deleted?
I.e., why can't the same solution work for both sets of files?

File: undocked/tools/export_program_info_sample/export_program_info_sample.vcxproj [hunk 9]
```diff
@@ -243,4 +243,4 @@ $(OutputPath)export_program_info_sample.exe</Command>
     </ProjectReference>
   </ItemGroup>
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
-</Project>
+</Project>
```
Comment (dthaler) [inline undocked/tools/export_program_info_sample/export_program_info_sample.vcxproj:246]: Why this change?

File: scripts/setup_build/setup_build.vcxproj.filters (hunk 5 above)
Comment (Alan-Jowett) [reply scripts/setup_build/setup_build.vcxproj.filters]: Editor stripped new lines from end of file for some reason.

File: tools/bpftool/bpftool.vcxproj (hunk 7 above)
Comment (Alan-Jowett) [reply tools/bpftool/bpftool.vcxproj]: Editor stripped new lines from end of file for some reason.

File: undocked/tools/export_program_info_sample/export_program_info_sample.vcxproj (hunk 9 above)
Comment (Alan-Jowett) [reply undocked/tools/export_program_info_sample/export_program_info_sample.vcxproj]: Editor stripped new lines from end of file for some reason.

**Review by dthaler — APPROVED** [review]

**Review by matthewige — APPROVED** [review]
